### PR TITLE
frontend: ignore non-existent projects in dash

### PR DESCRIPTION
**Pull request by dschaller**
### Description

Dash: fix lookup of projects that do not exist in response.

notFound

### Testing Performed

manual

### CODE REVIEW

**Review by scarlettperry — APPROVED** [review]
tested locally & lgtm!

File: frontend/workflows/projectSelector/src/selector-reducer.tsx [hunk 1]
```diff
@@ -207,16 +208,39 @@ const selectorReducer = (state: State, action: Action): State => {
...
-      */
-      return { ...state, loading: false, error: action?.payload?.result };
+    case "HYDRATE_ERROR": {
+      /**
+       * TODO: do we want to handle the error state differently? For example, when we render the error on the UI,
+       * it won't disapper unless there's a successful API call or if the user refreshes the page. If a user performs other
+       * actions, such as use the toggle/checkbox/ etc. the error message will be still be on the page
+       */
+      const err = action?.payload?.result as ClutchError;
+      // if (grpcCodeToHttpCode(err?.code) !== 404) {
```
Comment (mikecutalo) [inline frontend/workflows/projectSelector/src/selector-reducer.tsx:218]: remove?
```suggestion
```

File: frontend/workflows/projectSelector/src/selector-reducer.tsx [hunk 2]
```diff
@@ -207,16 +208,39 @@ const selectorReducer = (state: State, action: Action): State => {
...
+      if (err.status.code !== 404) {
+        return { ...state, loading: false, error: action?.payload?.result };
+      }
+      const errorMsg = err.message;
+      const validState = { ...state };
+      let projects = validState[Group.PROJECTS];
+      if (errorMsg) {
+        // this message is guaranteed to follow this format from the server
+        const ERROR_FMT = /unable to find project: (.+)/;
```
Comment (mikecutalo) [inline frontend/workflows/projectSelector/src/selector-reducer.tsx:227]: Why do we need to parse the error message? Currently only one 404 case can be hit, when a project is not found. Is it too much to assume that if we see a 404 then this is the case?

Comment (dschaller) [reply frontend/workflows/projectSelector/src/selector-reducer.tsx:226]: We need to parse it to extract just the service name that was searched for in order to remove it from the state

Comment (mikecutalo) [reply frontend/workflows/projectSelector/src/selector-reducer.tsx:226]: Ah alright i see now thanks

**Review by mikecutalo — APPROVED** [review]

**Pull request merged**